hack/docker-integration-test.sh

```diff
@@ -14,18 +13,78 @@ assert_success() {
 	fi
 }
 
+get_volume_id() {
+	VOLUME_ID=$(curl -X POST "http://localhost:9116/v1/volumes/filters" \
+		-H "accept: application/json" -H "Content-Type:application/json" \
+		-d "{ \"locator\": { \"name\": \"$1\"}}" \
+		| jq .[] | jq .[0] -r)
+}
+
+assert_attached(){
+	ATTACH_PATH=$(curl -X GET "http://localhost:9116/v1/volumes/inspect/$VOLUME_ID" \
+		-H "accept:application/json" | jq  ."volume" | jq ."attach_path" | jq .[0] -r)
+
+	# ATTACH_PATH is null when the volume unmounted. If it is non-null,
+	# we know that the volume is mounted. 
+	#
+	# Here we can assert that the volume is mounted
+	if [ $1 == "true" ];
+	then
+		echo "Asserting volume attached: $ATTACH_PATH"	
+		if [ $ATTACH_PATH != "null" ];
+		then
+			return 0
+		else
+			cleanup
+			exit 1
+		fi
+	# Here we can assert that the volume is unmounted
+	else
+		echo "Asserting volume detached: $ATTACH_PATH"	
+		if [ $ATTACH_PATH == "null" ];
+		then
+			return 0
+		else
+			cleanup
+			exit 1
+		fi
+	fi
+}
+
+
 # Start OSD
-make install
-sudo $GOPATH/bin/osd -d --driver=name=nfs,server=127.0.0.1,path=/nfs --sdkport 9106 --sdkrestport 9116 &
+sudo $GOPATH/bin/osd -d --driver=name=fake --sdkport 9106 --sdkrestport 9116 &
 jobs -l
+sleep 3
 
 # Test & assert
-sudo docker volume create -d nfs -o size=1234 > .osd_integration_test_id
+VOLUME_NAME=$(sudo docker volume create -d fake -o size=1234)
```

Comment:
See if you can create a secret every time using the program openstorage-sdk-auth by using `go install`. Then pass it in to docker using `-o token=<...>`

Reply:
I've changed `docker volume create` to pass in `-o token=<...>`, but how will we be passing the token to other calls (rm, inspect, mount/unmount, etc)? Do these calls require auth as well? I don't see a way to pass in options to these calls:
```
volume inspect --help

Usage:	docker volume inspect [OPTIONS] VOLUME [VOLUME...]

Display detailed information on one or more volumes

Options:
  -f, --format string   Format the output using the given Go template
      --help            Print usage

docker volume rm --help

Usage:	docker volume rm [OPTIONS] VOLUME [VOLUME...]

Remove one or more volumes

Aliases:
  rm, remove

Examples:

$ docker volume rm hello
hello


Options:
  -f, --force   Force the removal of one or more volumes
      --help    Print usage
```